File: spring-boot-configuration-properties-demo/src/main/java/com/github/jccode/configurationpropertiesdemo/MyAppConfigProperties.java
package com.github.jccode.configurationpropertiesdemo;

import org.springframework.boot.context.properties.ConfigurationProperties;

import java.util.List;
import java.util.Map;

@ConfigurationProperties(prefix = "app")
public class MyAppConfigProperties {

    private Map<String, Integer> thresholds;

    private Integer count;

    private List<String> list;

    public Map<String, Integer> getThresholds() {
        return thresholds;
    }

    public void setThresholds(Map<String, Integer> thresholds) {
        this.thresholds = thresholds;
    }

    public Integer getCount() {
        return count;
    }

    public void setCount(Integer count) {
        this.count = count;
    }

    public List<String> getList() {
        return list;
    }

    public void setList(List<String> list) {
        this.list = list;
    }

    @Override
    public String toString() {
        return "MyAppConfigProperties{" +
                "thresholds=" + thresholds +
                ", count=" + count +
                ", list=" + list +
                '}';
    }
}

File: spring-boot-configuration-properties-demo/src/main/scala/com/github/jccode/configurationpropertiesdemo/MyAppConfigProperties.scala
package com.github.jccode.configurationpropertiesdemo

import org.springframework.boot.context.properties.ConfigurationProperties

import scala.beans.BeanProperty


@ConfigurationProperties(prefix = "app")
class MyAppConfigProperties {

  @BeanProperty var thresholds: java.util.Map[String, Integer] = null

  @BeanProperty var count: Integer = null

  @BeanProperty var list: java.util.List[String] = null

  override def toString = s"MyAppConfigProperties($thresholds, $count, $list)"

}
